sechub-scan/src/test/java/com/mercedesbenz/sechub/domain/scan/ScanTypeCountTest.java

```diff
@@ -0,0 +1,123 @@
+// SPDX-License-Identifier: MIT
+package com.mercedesbenz.sechub.domain.scan;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import com.mercedesbenz.sechub.commons.model.ScanType;
+
+public class ScanTypeCountTest {
```

Comment:
I am missing tests which test incrementing the counter by more than one time (e.g. 101 times or some other value).

In addition, it would be interesting to have a test which sets the counter to Integer.MAX - 1 and count it up one or two and see what happens.

Reply:
@Jeeppler 

> In addition, it would be interesting to have a test which sets the counter to Integer.MAX - 1 and count it up one or two and see what happens.

I suggest this test, but I'm not sure this is what you really want.

```suggestion
    @Test
    void what_if_lowSeverityCount_equals_MAX_VALUE_and_increse_it() {
        /* prepare */
        for (int i = 0; i < Integer.MAX_VALUE; i++) {
            scanTypeCount.incrementLowSeverityCount();
        }

        /* execute */
        scanTypeCount.incrementLowSeverityCount();

        /* test */
        assertTrue(0 > scanTypeCount.getLowSeverityCount());
    }
```

Reply:
One test which increments several times. For example, 3+. Just to ensure the increment works with more than just one iteration.

In the second test you set the severityhighcount directly to Integer.MAX_VALUE - 1. Increment by one and by two to see what happens. In tests you can access the properties (fields) of a class directly. However, you might have to make the fields `protected/package private` rather than `private`.

Reply:
Done